src/core/Node.hack

```diff
@@ -11,7 +11,7 @@ use type Facebook\TypeAssert\IncorrectTypeException;
 use namespace Facebook\TypeAssert;
 use namespace HH\Lib\{C, Dict, Keyset, Str, Vec};
 
-abstract xhp class x:composable_element extends :xhp {
+abstract xhp class x:node extends :xhp {
```

Comment:
maybe add a docblock since the name is not as descriptive now

Reply:
> good luck in your ongoing war against hyphens and underscores

don't forget the colons!

Reply:
```
XHPChild
| |
| \- string, vec<XHPChild> etc
|
XHPRoot
|
Node
|  |
|  \- Primitive (directly implements `toStringAsync()`)
| 
Element (returns any XHPRoot, eventually reduces to primitives)
```

^ this is the basic type system


I'll put up a diff to kill XHPRoot too - AIUI every SHPRoot is a node, and every node is an XHPRoot

Reply:
... and perhaps :xhp and :x:node should be merged. Not seeing the need for separation any more